apps/internal/oauth/ops/accesstokens/tokens.go

```diff
@@ -173,14 +174,80 @@ type TokenResponse struct {
 	FamilyID       string                    `json:"foci"`
 	IDToken        IDToken                   `json:"id_token"`
 	ClientInfo     ClientInfo                `json:"client_info"`
-	ExpiresOn      internalTime.DurationTime `json:"expires_in"`
+	ExpiresOn      internalTime.DurationTime `json:"-"`
 	ExtExpiresOn   internalTime.DurationTime `json:"ext_expires_in"`
 	GrantedScopes  Scopes                    `json:"scope"`
 	DeclinedScopes []string                  // This is derived
 
 	AdditionalFields map[string]interface{}
+	scopesComputed   bool
+}
+
+func (tr *TokenResponse) UnmarshalJSON(data []byte) error {
+	type Alias TokenResponse
+	aux := &struct {
+		ExpiresIn json.Number `json:"expires_in,omitempty"`
+		ExpiresOn interface{} `json:"expires_on,omitempty"`
+		*Alias
+	}{
+		Alias: (*Alias)(tr),
+	}
+
+	// Unmarshal the JSON data into the aux struct
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	parseDuration := func(num json.Number) (int64, error) {
```

Comment:
How is this better than `DurationTime.UnmarshalJSON` i.e., what do you gain by unmarshaling `expires_in` to `json.Number` instead of `DurationTime`?

Reply:
shifted this to `internalTime.DurationTime`